pkg/plugins/helm/provider.go

```diff
@@ -53,37 +80,424 @@ func (p *Provider) setupHelm(kubeClient *kubernetes.Clientset, restClientConfig
+			if err != nil {
+				//network or api error, not considered task fail.
+				return false, nil
+			}
+
+			if resp.Info.Status.Code == release.Status_DEPLOYED {
+				return true, nil
+			}
+		case constants.ActionDeleteClusters:
+			resp, err := hc.ReleaseStatus(taskDirective.ClusterId)
+			if err != nil {
+				//network or api error, not considered task fail.
+				return false, nil
+			}
+
+			if resp.Info.Status.Code == release.Status_DELETED {
+				return true, nil
+			}
+		case constants.ActionCeaseClusters:
+			_, err := hc.ReleaseStatus(taskDirective.ClusterId)
+			if err != nil {
+				if _, ok := err.(transport.ConnectionError); ok {
+					return false, nil
+				}
+				return true, nil
+			}
+		}
+		return false, nil
+	}, timeout, waitInterval)
+
 	return nil
 }
+
 func (p *Provider) DescribeSubnet(runtimeId, subnetId string) (*models.Subnet, error) {
 	return nil, nil
 }
 func (p *Provider) DescribeVpc(runtimeId, vpcId string) (*models.Vpc, error) {
 	return nil, nil
 }
 
+func (p *Provider) getKubePodsAsClusterNodes(runtimeId, namespace, clusterId, owner string) ([]*pb.ClusterNode, error) {
+	kubeClient, _, err := p.getKubeClient(runtimeId)
+	if err != nil {
+		return nil, err
+	}
+
+	podsClient := kubeClient.CoreV1().Pods(namespace)
+	list, err := podsClient.List(metav1.ListOptions{LabelSelector: fmt.Sprintf("release=%s", clusterId)})
+	if err != nil {
+		return nil, err
+	}
+
+	var pbClusterNodes []*pb.ClusterNode
+	for _, v := range list.Items {
+
+		clusterNode := &models.ClusterNode{
+			NodeId:         models.NewClusterNodeId(),
+			ClusterId:      clusterId,
+			Name:           v.GetName(),
+			InstanceId:     string(v.GetUID()),
+			PrivateIp:      v.Status.PodIP,
+			Status:         string(v.Status.Phase),
+			Owner:          owner,
+			GlobalServerId: v.Spec.NodeName,
+			CustomMetadata: getLabelString(v.GetObjectMeta().GetLabels()),
+			CreateTime:     v.GetObjectMeta().GetCreationTimestamp().Time,
+			StatusTime:     v.GetObjectMeta().GetCreationTimestamp().Time,
+		}
+
+		pbClusterNode := models.ClusterNodeToPb(clusterNode)
+		pbClusterNodes = append(pbClusterNodes, pbClusterNode)
+	}
+
+	return pbClusterNodes, nil
+}
+
+func (p *Provider) UpdateClusterStatus(job *models.Job) error {
+	clusterWrapper, err := models.NewClusterWrapper(job.Directive)
+	if err != nil {
+		return err
+	}
+
+	runtimeId := clusterWrapper.Cluster.RuntimeId
+
+	ctx := context.Background()
+	runtimeClient, err := runtimeclient.NewRuntimeManagerClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	req := &pb.DescribeRuntimesRequest{RuntimeId: []string{runtimeId}}
+	resp, err := runtimeClient.DescribeRuntimes(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.RuntimeSet) == 0 {
+		return fmt.Errorf("Runtime [%s] not found", runtimeId)
+	}
+
+	runtime := resp.RuntimeSet[0]
+	namespace := runtime.GetZone().GetValue()
+
+	pbClusterNodes, err := p.getKubePodsAsClusterNodes(runtimeId, namespace, job.ClusterId, job.Owner)
```

Comment:
Modify use update sql, can not insert new rows. 
We may need a "INSERT ON DUPLICATE KEY UPDATE" func in pkg/db. 
@zheng1 Is there such func in dbr?